### PR TITLE
Modify bits and pieces of Gradle to accommodate Plugin Publish Plugin v1.0.0

### LINKED ISSUE #20142: publish using maven-publish resulting in shrinked pom.xml version
Hello!

Publishing my gradle plugin using maven-publish plugin to maven central repository producing shrieked pom.xml version. This shrieked version only contains artifact information and dependencies which is leading to fail publication on maven central.

Replacing maven-publish plugin with:

```
    dependencies {
        classpath 'com.vanniktech:gradle-maven-publish-plugin:0.19.0'
    }

apply plugin: "com.vanniktech.maven.publish"
``` 

Allows me to publish to maven central, since this plugin crates not filtered pom.xml.

For some reason output of maven-publish generatePomFileForPublication - is correct. Only problem with maven-publish it filter pom.xml when uploading to repository or local maven repo (looks like it also failed to update signatures, so pom file is unsigned in the result)

My projects are open source:

* https://gitlab.com/axet/gradle-android-dx

To make 'com.vanniktech.maven.publish' plugin to work update gradle.wrapper to the 7.0

You can see pom.xml here:

* https://plugins.gradle.org/m2/com/github/axet/gradle-android-dx/0.0.6/

### PR DESCRIPTION
**Pull request by jbartok**
Also fixes #20142

### CODE REVIEW

**Comment by jbartok** [comment]
@bot-gradle test QFL

**Comment by jbartok** [comment]
@bot-gradle test QFL

File: subprojects/docs/src/docs/userguide/developing-plugins/publishing_gradle_plugins.adoc
Comment (ljacomet) [inline subprojects/docs/src/docs/userguide/developing-plugins/publishing_gradle_plugins.adoc:125]: I do not understand this change. I thought the plan was to have all plugin metadata in the `gradlePlugins` block

File: subprojects/plugin-development/src/main/java/org/gradle/plugin/devel/plugins/IvyPluginPublishingPlugin.java
Comment (ljacomet) [inline subprojects/plugin-development/src/main/java/org/gradle/plugin/devel/plugins/IvyPluginPublishingPlugin.java:65]: I do not think we can just do that, it would impact anyone, including those using the plugin publish plugin aith a version before 1.0

File: subprojects/plugin-development/src/main/java/org/gradle/plugin/devel/GradlePluginDevelopmentExtension.java
Comment (ljacomet) [inline subprojects/plugin-development/src/main/java/org/gradle/plugin/devel/GradlePluginDevelopmentExtension.java:169]: Invoking this method should print a deprecation warning as well. The text should also say that it is effectively ignored when using the plugin publish >= 1.0

**Review by ljacomet — CHANGES_REQUESTED** [review]
Looks good overall

However the deprecation logic needs to be adapted, see below:

File: subprojects/docs/src/docs/userguide/developing-plugins/publishing_gradle_plugins.adoc
Comment (jbartok) [reply subprojects/docs/src/docs/userguide/developing-plugins/publishing_gradle_plugins.adoc:125]: I've made the change to illustrate that the description can also be specified globally, for the entire bundle, not just individually for each published plugin. Do we want to deprecate that capability?

**Comment by jbartok** [comment]
@bot-gradle test QFL

**Comment by jbartok** [comment]
@bot-gradle test QFL

**Comment by jbartok** [comment]
@bot-gradle test QFL

**Comment by jbartok** [comment]
@bot-gradle test

**Review by ljacomet — CHANGES_REQUESTED** [review]
Few remarks on the documentation, see below

File: subprojects/docs/src/docs/userguide/developing-plugins/publishing_gradle_plugins.adoc
Comment (jbartok) [reply subprojects/docs/src/docs/userguide/developing-plugins/publishing_gradle_plugins.adoc:111]: I'll set a reminder to do this once 1.0.0 has been out for a while and I can find such a plugin.

**Comment by jbartok** [comment]
@bot-gradle test RFM

**Comment by jbartok** [comment]
@bot-gradle test RFM

**Comment by jbartok** [comment]
@bot-gradle test RFM

**Comment by jbartok** [comment]
@bot-gradle test RFM

**Comment by jbartok** [comment]
@bot-gradle test RFM

**Review by ljacomet — CHANGES_REQUESTED** [review]

**Comment by jbartok** [comment]
@bot-gradle test RFM

**Comment by jbartok** [comment]
@bot-gradle test RFM

**Comment by jbartok** [comment]
@bot-gradle test RFM

File: subprojects/docs/src/docs/userguide/core-plugins/java_gradle_plugin.adoc
Comment (ljacomet) [inline subprojects/docs/src/docs/userguide/core-plugins/java_gradle_plugin.adoc:18]: Let's enforce one sentence per line as soon as we touch an asciidoc section.

File: subprojects/docs/src/docs/userguide/core-plugins/java_gradle_plugin.adoc
Comment (ljacomet) [inline subprojects/docs/src/docs/userguide/core-plugins/java_gradle_plugin.adoc:34]: same here

**Review by ljacomet — APPROVED** [review]
LGTM!

Minor edits to docs, see below. Can be merged once changed, no need for another review.

**Comment by jbartok** [comment]
@bot-gradle test and merge

**Pull request merged**
